Include/internal/pycore_bitutils.h

```diff
@@ -131,8 +131,32 @@ _Py_popcount32(uint32_t x)
 }
 
 
+// Return the index of the most significant 1 bit in 'x'. This is the smallest
+// integer k such that x < 2**k. Equivalent to floor(log2(x)) + 1 for x != 0.
+static inline int
+_Py_bit_length(unsigned long x) {
+    if (!x) {
+        return 0;
+    }
+#if (defined(__clang__) || defined(__GNUC__))
+    return sizeof(unsigned long) * 8 - __builtin_clzl(x);
+#elif defined(_MSC_VER)
+    Py_BUILD_ASSERT(4 == sizeof(unsigned long));
+    unsigned long msb;
+    _BitScanReverse(&msb, x);
+    return 32 - msb;
```

Comment:
In PR #594 you wrote a different implementation:

```
    unsigned long idx;
    if (_BitScanReverse(&idx, d))
        return idx + 1;
    else
        return 0;
```

Reply:
Thanks. I misremembered (`_BitScanReverse` counts from the opposite side that clz does). This also explains the broken Windows build.